src/chapters/typescript/classes.rst

```diff
@@ -0,0 +1,99 @@
+Classes and Interfaces in TypeScript
+====================================
+
+Classes in TypeScript look something like this:
+
+.. sourcecode:: js
+   :linenos:
+
+   class Astronaut {
+      name: string;
+      constructor(firstName: string, lastName: string) {
+         this.name = firstName + " " + lastName;
+      }
+      greet() {
+         return "Hello, " + this.name;
+      }
+   }
+
+   let Bob = Astronaut("Bob","Smith");
+
+You may remember the ``this`` and ``new`` keywords from working with classes in JavaScript.
+Earlier in the chapter, we also noted that when declaring variables in TypeScript, we have to specify the type of value.
+The same applies to function parameters, as you can see in the constructor.
+
+Classes in TypeScript can also use the ``extends`` keyword to denote child and parent classes, as shown here:
+
+.. sourcecode:: js
+   :linenos:
+
+   class Panthera {
+      roar: string;
+      constructor(currentRoar: string) {
+         this.roar = currentRoar;
+      }
+   }
+
+   class Tiger extends Panthera {
+      stripes: boolean = true;
+
+   }
+
+   let tigger = new Tiger("loud");
+   console.log(tigger.roar);
+   console.log(tigger.stripes);
+
+
+``export`` is used to make the class available in the global scope.
+
+Interfaces
+----------
+
+When we start working with Angular, you may see the ``interface`` keyword quite a bit. Interfaces in TypeScript name types that are created by the programmer.
```

Comment:
What do you think about adding something like this?  "Like classes, interfaces define properties and methods that a type will have. The difference is that interfaces do NOT include initialization of properties or implementations of method. Interfaces define the contract that other classes or objects must comply with if they declare that they implement that interface. Interfaces are often implemented by multiple classes, that flexility allows different classes to be seen as a common type. That common contract is useful when certain functionality is needed by a function parameter."

Reply:
I reworded this paragraph because it is a little confusing. Hope it still captures the intended spirit!